### PR TITLE
Label API to use `extraClasses` and Label Unit with @dojo/test-extras

### LINKED ISSUE #139: Label: Update Unit Tests
Update Label unit tests to use `test-extras`.

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Supersedes the existing PR #167 that required conflict resolution and adding the `Label` changes to `Timepicker`.

I _think_ this is what is currently required with `Label`, to use `extraClasses` that get passed from the parent widget - but this needs to be discussed between @tom, @mwistrand, @smhigley and @bitpshr about whether this is the right approach.

For now I am proposing we take this PR so we can unblock resolving the rest of the outstanding widget unit test PRs.... 

Resolves #139 

Closes: #167

### CODE REVIEW

File: src/checkbox/Checkbox.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export default class Checkbox extends CheckboxBase<CheckboxProperties> {
 
 		if (label) {
 			checkboxWidget = w(Label, {
-				classes: this.classes(css.root, ...stateClasses),
+				extraClasses: { root: parseLabelClasses(this.classes(css.root, ...stateClasses).get()) },
 				formId,
 				label
 			}, children);
```
Comment (tomdye) [inline src/checkbox/Checkbox.ts]: pass `theme`

File: src/radio/Radio.ts [hunk 2]
```diff
@@ -118,7 +118,7 @@ export default class Radio extends RadioBase<RadioProperties> {
 
 		if (label) {
 			radioWidget = w(Label, {
-				classes: this.classes(css.root, ...stateClasses),
+				extraClasses: { root: parseLabelClasses(this.classes(css.root, ...stateClasses).get()) },
 				formId,
 				label
```
Comment (tomdye) [inline src/radio/Radio.ts]: pass `theme`

File: src/slider/Slider.ts [hunk 3]
```diff
@@ -183,7 +183,7 @@ export default class Slider extends SliderBase<SliderProperties> {
 
 		if (label) {
 			sliderWidget = w(Label, {
-				classes: this.classes(css.root, ...stateClasses).fixed(css.rootFixed),
+				extraClasses: { root: parseLabelClasses(this.classes(css.root, ...stateClasses).fixed(css.rootFixed)()) },
 				formId,
 				label
```
Comment (tomdye) [inline src/slider/Slider.ts]: pass `theme`

File: src/textarea/Textarea.ts [hunk 4]
```diff
@@ -152,7 +152,7 @@ export default class Textarea extends TextareaBase<TextareaProperties> {
 
 		if (label) {
 			textareaWidget = w(Label, {
-				classes: this.classes(css.root, ...stateClasses),
+				extraClasses: { root: parseLabelClasses(this.classes(css.root, ...stateClasses)()) },
 				formId,
 				label
```
Comment (tomdye) [inline src/textarea/Textarea.ts]: pass `theme`

File: src/textinput/TextInput.ts [hunk 5]
```diff
@@ -150,7 +150,7 @@ export default class TextInput extends TextInputBase<TextInputProperties> {
 
 		if (label) {
 			textinputWidget = w(Label, {
-				classes: this.classes(css.root, ...stateClasses),
+				extraClasses: { root: parseLabelClasses(this.classes(css.root, ...stateClasses)()) },
 				formId,
 				label
```
Comment (tomdye) [inline src/textinput/TextInput.ts]: pass `theme`

File: src/timepicker/TimePicker.ts [hunk 6]
```diff
@@ -184,12 +184,13 @@ export class TimePicker extends TimePickerBase<TimePickerProperties> {
...
+						css.input,
 						disabled ? css.disabled : null,
 						invalid ? css.invalid : null,
 						readOnly ? css.readonly : null,
-						required ? css.required : null
-					),
+						required ? css.required : null).get())
+					},
 					formId,
 					label
```
Comment (tomdye) [inline src/timepicker/TimePicker.ts]: pass `theme`

**Comment by agubler** [comment]
@tomdye feedback addressed.

**Pull request merged**